openedx/core/djangoapps/programs/tasks.py

```diff
@@ -782,39 +783,44 @@ def update_certificate_available_date_on_course_update(self, course_key):
-    # Update the Credentials service's CourseCertificate configuration with the new `certificate_available_date` if:
-    #   - The course-run is instructor-paced, AND
-    #   - The `certificates_display_behavior` is set to "end_with_date",
-    if (
-        course_overview
-        and course_overview.self_paced is False
-        and course_overview.certificates_display_behavior == CertificatesDisplayBehaviors.END_WITH_DATE
-    ):
-        LOGGER.info(
-            f"Queueing task to update the `certificate_available_date` of course-run {course_key} to "
-            f"[{course_overview.certificate_available_date}] in the Credentials service"
-        )
-        update_credentials_course_certificate_configuration_available_date.delay(
-            str(course_key), str(course_overview.certificate_available_date)
-        )
-    # OR,
-    #   - The course-run is self-paced, AND
-    #   - The `certificate_available_date` is (now) None. (This task will be executed after an update to the course
-    #     overview)
-    # There are times when the CourseCertificate configuration of a self-paced course-run in Credentials can become
-    # associated with a `certificate_available_date`. This ends up causing learners' certificate to be incorrectly
-    # hidden. This is due to the Credentials IDA not understanding the concept of course pacing. Thus, we need a way
-    # to remove this value from self-paced courses in Credentials.
-    elif course_overview and course_overview.self_paced is True and course_overview.certificate_available_date is None:
-        LOGGER.info(
-            "Queueing task to remove the `certificate_available_date` in the Credentials service for course-run "
-            f"{course_key}"
-        )
+    # When updating the certificate available date of instructor-paced course runs,
+    #   - If the display behavior is set to "A date after the course end date" (END_WITH_DATE), we should send the
+    #     certificate available date set by the course team in Studio (and stored as part of the course runs Course
+    #     Overview)
+    #   - If the display behavior is set to "End date of course" (END), we should send the end date of the course run
+    #     as the certificate available date. We send the end date because the Credentials IDA doesn't understand the
+    #     concept of course pacing and needs an explicit date in order to correctly gate the visibility of course and
+    #     program certificates.
+    #   - If the display behavior is set to "Immediately upon passing" (EARLY_NO_INFO), we should always send None for
+    #     the course runs certificate available date. A course run configured with this display behavior must not have a
+    #     certificate available date associated with or the Credentials system will incorrectly hide certificates from
+    #     learners.
+    course_overview = get_course_overview_or_none(course_key)
+    if (course_overview and course_overview.self_paced is False):
```

Comment:
I would probably just put 

```if not  course_overview: 
    logger.WARNING ("from line 824")
    return
```  
on the line above this, so you don't have to check for its existence multiple times.

And then actually if you move the new behavior on 820-821 up to here, immediately after that conditional, then you don't need this line at all, it's all just `else` from there.